examples/applications/plot_out_of_core_classification.py

```diff
@@ -214,44 +227,67 @@ def progress(blocknum, bs, size):
     'Passive-Aggressive': PassiveAggressiveClassifier(tol=1e-3),
 }
 
+# test data statistics
+test_stats = {'n_test': 0, 'n_test_pos': 0}
 
-def get_minibatch(doc_iter, size, pos_class=positive_class):
+
+def get_testdata(doc_iter_test, pos_class=positive_class):
     """Extract a minibatch of examples, return a tuple X_text, y.
 
     Note: size is before excluding invalid docs with no topics assigned.
 
     """
-    data = [(u'{title}\n\n{body}'.format(**doc), pos_class in doc['topics'])
-            for doc in itertools.islice(doc_iter, size)
-            if doc['topics']]
-    if not len(data):
-        return np.asarray([], dtype=int), np.asarray([], dtype=int)
-    X_text, y = zip(*data)
-    return X_text, np.asarray(y, dtype=int)
 
+    data_test = [(
+        u'{title}\n\n{body}'.format(**doc), pos_class in doc['topics'])
+        for doc in doc_iter_test
+        if doc['topics']]
 
-def iter_minibatches(doc_iter, minibatch_size):
-    """Generator of minibatches."""
-    X_text, y = get_minibatch(doc_iter, minibatch_size)
-    while len(X_text):
-        yield X_text, y
-        X_text, y = get_minibatch(doc_iter, minibatch_size)
+    if not len(data_test):
+        return np.asarray([], dtype=int), np.asarray([], dtype=int)
 
+    X_TextTest, y_test = zip(*data_test)
 
-# test data statistics
-test_stats = {'n_test': 0, 'n_test_pos': 0}
+    return X_TextTest, np.asarray(y_test, dtype=int)
 
-# First we hold out a number of examples to estimate accuracy
-n_test_documents = 1000
 tick = time.time()
-X_test_text, y_test = get_minibatch(data_stream, 1000)
+X_TextTest, y_test = get_testdata(data_stream_test, positive_class)
 parsing_time = time.time() - tick
+
+# Vectorized Test Data
 tick = time.time()
-X_test = vectorizer.transform(X_test_text)
+X_test = vectorizer.transform(X_TextTest)
 vectorizing_time = time.time() - tick
-test_stats['n_test'] += len(y_test)
-test_stats['n_test_pos'] += sum(y_test)
-print("Test set is %d documents (%d positive)" % (len(y_test), sum(y_test)))
+
+test_stats['n_test'] = len(y_test)
+test_stats['n_test_pos'] = sum(y_test)
+n_test_documents = len(y_test)
+
+
+def get_minibatch(doc_iter_train, size, pos_class=positive_class):
+    """Extract a minibatch of examples, return a tuple X_text, y.
+
```

Comment:
IMO would be great write the Parameters and returns in the format used on sklearn docs. I mean:

```
"""Extract a minibatch of examples

Parameters
---------------
...

Return
---------
.....

"""

```

Reply:
Thanks